PopPUNK/web.py

```diff
@@ -0,0 +1,230 @@
+        core_only = False
+        accessory_only = False
+        assign_lineage = True
+        rank = 1
+        exact_count = False
+        web = True
+        
+        #Options required by setupDBFuncs()
+        gpu_sketch = False
+        deviceid = 0
+        gpu_dist = False
+        min_kmer_count = 0
+        min_k = 14
+        max_k = 29
+        k_step = 3
+        use_accessory = False
+        use_mash = False
+        codon_phased = False
+        microreact = False
+        cytoscape = False
+        overwrite = True
+        phandango = False
+        grapetree = False
+        info_csv = None
+        rapidnj = None
+        perplexity = 20.0
+        existing_scheme = None
+        rank_list = None
+        sketch_sizes = 10000
+        no_stream = False
+        mash = None
+
+    return (args)
+
+def get_colours(query, clusters):
+    """Colour array for Plotly.js"""
+    colours = []
+    query = '"' + str(query) + '"'
+    for clus in clusters:
+        if not clus == query:
+            colours.append('"blue"')
+        else:
+            colours.append('"rgb(255,128,128)"')
+    return colours
+
+def sketch_to_hdf5(sketch, output):
+    """Convert JSON sketch to query hdf5 database"""
+    kmers = []
+    dists = []
+
+    sketch_dict = json.loads(sketch)
+    qNames = ["query"]
+    queryDB = h5py.File(os.path.join(output, output + '.h5'), 'w')
+    sketches = queryDB.create_group("sketches")
+    sketch_props = sketches.create_group(qNames[0])
+
+    for key, value in sketch_dict.items():
+        try:
+            kmers.append(int(key))
+            dists.append(np.array(value))
+        except:
+            if key == "bases":
+                sketch_props.attrs['base_freq'] = value
+            elif key == "bbits":
+                sketch_props.attrs['bbits'] = value
+            elif key == "length":
+                sketch_props.attrs['length'] = value
+            elif key == "missing_bases":
+                sketch_props.attrs['missing_bases'] = value
+            elif key == "sketchsize64":
+                sketch_props.attrs['sketchsize64'] = value
+            elif key == "version":
+                sketches.attrs['sketch_version'] = value
+            else:
+                raise AttributeError(key + " Not recognised")
+    
+    sketches.attrs['codon_phased'] = 0
+    sketch_props.attrs['kmers'] = kmers
+    for k_index in range(len(kmers)):
+        k_spec = sketch_props.create_dataset(str(kmers[k_index]), data=dists[k_index], dtype='uint64')
+        k_spec.attrs['kmer-size'] = kmers[k_index]
+    queryDB.close()
+    return qNames
+
+def ReformatNode(node_list):
+    """Graphml node to JSON"""
+    json_nodes = []
+    for node in node_list:
+        id = node.split('"')[1]
+        label = re.search('<data key="key0">(.*)</data>', node).group(1)
+        data = '{"data": {"id":"' + id + '", "label":"' + label + '"}}'
+        json_nodes.append(data)
+    return json_nodes
```

Comment:
I know this isn't finished yet, but is using re as robust as saving and loading via an intermediate format? Did the code come from somewhere?